src/main.rs

```diff
@@ -55,61 +64,82 @@ fn main() -> Result<(), String> {
     let escape_keycode = find_escape_keycode(&conn);
     grab_key(&conn, root, escape_keycode);
 
-    let screen_rect =
-        xcb::Rectangle::new(0, 0, screen.width_in_pixels(), screen.height_in_pixels());
+    let screen_rect = xproto::Rectangle {
+        x: 0,
+        y: 0,
+        width: screen.width_in_pixels,
+        height: screen.height_in_pixels,
+    };
 
     // TODO event handling for expose/keypress
-    let values = [
-        // ?RGB. First 4 bytes appear to do nothing
-        (xcb::CW_BACK_PIXEL, line_colour),
-        (
-            xcb::CW_EVENT_MASK,
-            xcb::EVENT_MASK_EXPOSURE
-            | xcb::EVENT_MASK_KEY_PRESS // we'll need this later
-            | xcb::EVENT_MASK_STRUCTURE_NOTIFY
-            | xcb::EVENT_MASK_SUBSTRUCTURE_NOTIFY,
-        ),
-        (xcb::CW_OVERRIDE_REDIRECT, 1u32), // Don't be window managed
-    ];
-
-    xcb::create_window(
+    let value_list = xproto::CreateWindowAux::new()
+        .background_pixel(line_colour)
+        .event_mask(
+            xproto::EventMask::Exposure
+                | xproto::EventMask::KeyPress
+                | xproto::EventMask::StructureNotify
+                | xproto::EventMask::SubstructureNotify,
+        )
+        .override_redirect(1);
```

Comment:
Comment was lost here? "Don't be window managed"

Reply:
The `?RGB` comment was also lost. Keypress comment isn't needed anymore since we do use the event. What does it mean to be window managed?

Reply:
that the window manager doesn't manage this window